pkg/endpoint/policy.go

```diff
@@ -680,6 +516,15 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 		"policyRevision.next": e.nextPolicyRevision,
 	}).Debug("Done regenerating")
 
+	// If no policy or options change occurred for this endpoint then the
+	// endpoint is already running the latest revision; otherwise we have to
+	// wait for the regeneration of the endpoint to complete. This allows us
+	// to signal that regeneration of BPF is not necessary.
+	if !policyChanged && !optsChanged {
+		e.setPolicyRevision(revision)
```

Comment:
Should this be somehow updated within the `syncPolicyMapController` instead? Otherwise we're signalling that we've updated the policy revision, but haven't guaranteed that the new policy revision is implemented yet.

Reply:
Hm, I see what you're saying here. I think the key thing here is that we need to track revisions for each part of the datapath (which we do for proxy). Similarly, we need the 'policymap revision'